sdk/tables/Azure.Data.Tables/tests/TableClientQueryExpressionTests.cs

```diff
@@ -213,9 +214,36 @@ public void TestDictionaryTableEntityFilterExpressions(string expectedFilter, Ex
 
         [TestCaseSource(nameof(UnSupportedTableItemExpressionTestCases))]
         [Test]
+        [NonParallelizable]
         public void TestTableItemFilterExpressionsUnsupported(Expression<Func<TableEntity, bool>> expression)
         {
             Assert.Throws<NotSupportedException>(() => TableClient.CreateQueryFilter(expression));
         }
+
+        [TestCaseSource(nameof(UnSupportedTableItemExpressionTestCases))]
+        [Test]
+        [NonParallelizable]
+        public void TestTableItemFilterExpressionsUnsupportedDoesNotThrowWithCompatSwitch(Expression<Func<TableEntity, bool>> expression)
+        {
+            if (expression == s_TEequalsStaticUnsupported)
```

Comment:
```suggestion
            if (expression == s_TEqualsStaticUnsupported)
```

Same here.  😄

Reply:
This one is on purpose 😄 TE == TableEntity